### ISSUE TITLE
@capacitor/app : error: invalid source release: 17

### ISSUE THREAD

**Issue by sys31337** [ionic-team/capacitor-plugins#1580]
### Plugin(s)
@capacitor/app
@capacitor/app-launcher

### Capacitor Version

```
Capacitor Doctor

Latest Dependencies:

  @capacitor/cli: 5.0.1
  @capacitor/core: 5.0.1
  @capacitor/android: 5.0.1
  @capacitor/ios: 5.0.1

Installed Dependencies:

  @capacitor/cli: 4.8.0
  @capacitor/android: 4.8.0
  @capacitor/ios: 4.8.0
  @capacitor/core: 4.8.0

[success] Android looking great! 👌
[error] Xcode is not installed```

### Platform(s)
Android



### Current Behavior
```
$ cap run android
√ Copying web assets from dist to android\app\src\main\assets\public in 357.73ms
√ Creating capacitor.config.json in android\app\src\main\assets in 2.14ms
[info] Found 1 Cordova plugin for android:
       cordova-plugin-file@7.0.0
| copy android [info] Inlining sourcemaps
√ copy android in 1.15s
√ Updating Android plugins in 12.08ms
[info] Found 2 Capacitor plugins for android:
       @capacitor/app@5.0.1
       @capacitor/app-launcher@5.0.1
| update android [info] Found 1 Cordova plugin for android:
       cordova-plugin-file@7.0.0
√ update android in 239.25ms
√ Please choose a target device: » Google sdk_gphone64_x86_64 (emulator-5554)
× Running Gradle build - failed!
[error] Starting a Gradle Daemon, 1 incompatible and 3 stopped Daemons could not be reused, use --status for
        details
        
        > Configure project :app
        WARNING:Using flatDir should be avoided because it doesn't support any meta-data formats.

        > Configure project :capacitor-cordova-android-plugins
        WARNING:Using flatDir should be avoided because it doesn't support any meta-data formats.

        > Task :app:preBuild UP-TO-DATE
        > Task :app:preDebugBuild UP-TO-DATE
        > Task :app:mergeDebugNativeDebugMetadata NO-SOURCE
        > Task :capacitor-android:preBuild UP-TO-DATE
        > Task :capacitor-android:preDebugBuild UP-TO-DATE
        > Task :capacitor-android:compileDebugAidl NO-SOURCE
        > Task :capacitor-app:preBuild UP-TO-DATE
        > Task :capacitor-app:preDebugBuild UP-TO-DATE
        > Task :capacitor-app:compileDebugAidl NO-SOURCE
        > Task :capacitor-app-launcher:preBuild UP-TO-DATE
        > Task :capacitor-app-launcher:preDebugBuild UP-TO-DATE
        > Task :capacitor-app-launcher:compileDebugAidl NO-SOURCE
        > Task :capacitor-cordova-android-plugins:preBuild UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:preDebugBuild UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:compileDebugAidl NO-SOURCE
        > Task :app:compileDebugAidl NO-SOURCE
        > Task :capacitor-android:packageDebugRenderscript NO-SOURCE
        > Task :capacitor-app:packageDebugRenderscript NO-SOURCE
        > Task :capacitor-app-launcher:packageDebugRenderscript NO-SOURCE
        > Task :capacitor-cordova-android-plugins:packageDebugRenderscript NO-SOURCE
        > Task :app:compileDebugRenderscript NO-SOURCE
        > Task :app:generateDebugBuildConfig UP-TO-DATE
        > Task :app:javaPreCompileDebug UP-TO-DATE
        > Task :capacitor-android:writeDebugAarMetadata UP-TO-DATE
        > Task :capacitor-app:writeDebugAarMetadata UP-TO-DATE
        > Task :capacitor-app-launcher:writeDebugAarMetadata UP-TO-DATE
        > Task :app:generateDebugResValues UP-TO-DATE
        > Task :app:generateDebugResources UP-TO-DATE
        > Task :capacitor-android:compileDebugRenderscript NO-SOURCE
        > Task :capacitor-android:generateDebugResValues UP-TO-DATE
        > Task :capacitor-android:generateDebugResources UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:writeDebugAarMetadata
        > Task :capacitor-android:packageDebugResources UP-TO-DATE
        > Task :capacitor-app:compileDebugRenderscript NO-SOURCE
        > Task :capacitor-app:generateDebugResValues UP-TO-DATE
        > Task :capacitor-app:generateDebugResources UP-TO-DATE
        > Task :capacitor-app:packageDebugResources UP-TO-DATE
        > Task :capacitor-app-launcher:compileDebugRenderscript NO-SOURCE
        > Task :capacitor-app-launcher:generateDebugResValues UP-TO-DATE
        > Task :capacitor-app-launcher:generateDebugResources UP-TO-DATE
        > Task :capacitor-app-launcher:packageDebugResources UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:compileDebugRenderscript NO-SOURCE
        > Task :capacitor-cordova-android-plugins:generateDebugResValues
        > Task :capacitor-cordova-android-plugins:generateDebugResources
        > Task :capacitor-cordova-android-plugins:packageDebugResources
        > Task :app:mergeDebugResources UP-TO-DATE
        > Task :app:checkDebugAarMetadata UP-TO-DATE
        > Task :app:mapDebugSourceSetPaths UP-TO-DATE
        > Task :app:createDebugCompatibleScreenManifests UP-TO-DATE
        > Task :app:extractDeepLinksDebug UP-TO-DATE
        > Task :capacitor-android:extractDeepLinksDebug UP-TO-DATE
        > Task :capacitor-android:processDebugManifest UP-TO-DATE
        > Task :capacitor-app:extractDeepLinksDebug UP-TO-DATE
        > Task :capacitor-app:processDebugManifest UP-TO-DATE
        > Task :capacitor-app-launcher:extractDeepLinksDebug UP-TO-DATE
        > Task :capacitor-app-launcher:processDebugManifest UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:extractDeepLinksDebug
        > Task :capacitor-android:compileDebugLibraryResources UP-TO-DATE
        > Task :capacitor-android:parseDebugLocalResources UP-TO-DATE
        > Task :capacitor-android:generateDebugRFile UP-TO-DATE
        > Task :capacitor-app:compileDebugLibraryResources UP-TO-DATE
        > Task :capacitor-app:parseDebugLocalResources UP-TO-DATE
        > Task :capacitor-app:generateDebugRFile UP-TO-DATE
        > Task :capacitor-app-launcher:compileDebugLibraryResources UP-TO-DATE
        > Task :capacitor-app-launcher:parseDebugLocalResources UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:processDebugManifest
        > Task :capacitor-app-launcher:generateDebugRFile UP-TO-DATE
        > Task :app:processDebugMainManifest UP-TO-DATE
        > Task :app:processDebugManifest UP-TO-DATE
        > Task :app:processDebugManifestForPackage UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:compileDebugLibraryResources
        > Task :capacitor-android:generateDebugBuildConfig UP-TO-DATE
        > Task :capacitor-android:javaPreCompileDebug UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:parseDebugLocalResources
        > Task :capacitor-android:compileDebugJavaWithJavac UP-TO-DATE
        > Task :capacitor-android:bundleLibCompileToJarDebug UP-TO-DATE
        > Task :capacitor-app:generateDebugBuildConfig UP-TO-DATE
        > Task :capacitor-app:javaPreCompileDebug UP-TO-DATE
        > Task :capacitor-cordova-android-plugins:generateDebugRFile
        > Task :capacitor-app:compileDebugJavaWithJavac FAILED

        FAILURE: Build failed with an exception.

        * What went wrong:
        Execution failed for task ':capacitor-app:compileDebugJavaWithJavac'.
        > error: invalid source release: 17

        * Try:
        > Run with --stacktrace option to get the stack trace.
        > Run with --info or --debug option to get more log output.
        > Run with --scan to get full insights.

        * Get more help at https://help.gradle.org

        BUILD FAILED50 actionable tasks: 9 executed, 41 up-to-date
        in 9s

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command
```

after installing this two plugins, using yarn add, and then I ran npx cap sync android the application failed to start,

**Comment by BalenDezai** [comment]
Got the same issue here. I tried changing my JDK version to 17 from 11 but it still persisted.

**Comment by sys31337** [comment]
I fixed mine aswell with version 5.0.1, just had to reinstall capacitor fully, upgraded my gradle and downloaded jdk 17

**Comment by jcesarmobile** [comment]
Version 5 of the plugins are meant to be used with Capacitor 5, but you are still using Capacitor 4.
Those errors are telling you the same thing.
[Migrate to Capacitor 5](https://capacitorjs.com/docs/updating/5-0#using-the-cli-to-migrate) or install version 4 of the plugins, i.e. `npm install @capacitor/app@^4`

I've documented that Capacitor 4 should use version 4 of the plugins
https://github.com/ionic-team/capacitor-plugins/pull/1582
